app/controllers/users_controller.rb

```diff
@@ -81,14 +81,21 @@ def new # TODO needs more rigorous testing
   def add_patient
     current_user.add_patient @patient
     respond_to do |format|
-      format.js { render template: 'users/refresh_patients', layout: false }
+      format.js { render template: 'users/refresh_patients', locals: {msg: 'Successfully added Patient'}, layout: false }
     end
   end
 
   def remove_patient
     current_user.remove_patient @patient
     respond_to do |format|
-      format.js { render template: 'users/refresh_patients', layout: false }
+      format.js { render template: 'users/refresh_patients', locals: {msg: 'Successfully removed Patient'}, layout: false }
+    end
+  end
+
+  def remove_all_patients
+    current_user.patients.destroy_all
```

Comment:
If memory serves, you want `clear` here rather than `destroy_all` (which I believe might actually actually destroy the patient records, unfortunately).